src/components/product/product.service.ts

```diff
@@ -370,105 +345,102 @@ export class ProductService {
         relation('out', 'produces', { active: true }),
         node('p', 'Producible', { active: true }),
       ])
       .return('p')
+      .asResult<{ p: Node<BaseNode> }>()
       .first();
 
-    if (!produces) {
-      const scriptureReferences = await this.listScriptureReferences(
-        baseNodeProps.id,
-        'Product',
-        session
-      );
+    const scriptureReferencesValue = await this.scriptureRefService.list(
+      id,
+      session,
+      { isOverriding: pr ? true : false }
+    );
+
+    if (!pr) {
       return {
-        ...baseNodeProps,
-        ...securedProperties,
+        ...parseBaseNodeProperties(result.node),
+        ...rest,
         scriptureReferences: {
-          canRead: result.canScriptureReferencesRead,
-          canEdit: result.canScriptureReferencesEdit,
-          value: scriptureReferences,
+          ...rest.scriptureReferences,
+          value: scriptureReferencesValue,
         },
         mediums: {
-          ...securedProperties.mediums,
-          value: securedProperties.mediums.value ?? [],
+          ...rest.mediums,
+          value: rest.mediums.value ?? [],
         },
         purposes: {
-          ...securedProperties.purposes,
-          value: securedProperties.purposes.value ?? [],
+          ...rest.purposes,
+          value: rest.purposes.value ?? [],
         },
       };
     }
 
-    const scriptureReferencesOverride = await this.listScriptureReferences(
-      baseNodeProps.id,
-      'Product',
-      session,
-      { isOverride: true }
-    );
-
-    const typeName = difference(produces.p.labels, [
-      'Producible',
-      'BaseNode',
-    ])[0];
+    const typeName = difference(pr.p.labels, ['Producible', 'BaseNode'])[0];
 
     const producible = await this.getProducibleByType(
-      produces.p.properties.id,
+      pr.p.properties.id,
       typeName,
       session
     );
 
     return {
-      ...baseNodeProps,
-      ...securedProperties,
+      ...parseBaseNodeProperties(result.node),
+      ...rest,
+      isOverriding,
       scriptureReferences: {
-        canRead: result.canScriptureReferencesRead,
-        canEdit: result.canScriptureReferencesEdit,
-        value: [],
+        ...rest.scriptureReferences,
+        value: !isOverriding.value
+          ? producible?.scriptureReferences.value
+          : scriptureReferencesValue,
       },
       mediums: {
-        ...securedProperties.mediums,
-        value: securedProperties.mediums.value ?? [],
+        ...rest.mediums,
+        value: rest.mediums.value ?? [],
       },
       purposes: {
-        ...securedProperties.purposes,
-        value: securedProperties.purposes.value ?? [],
+        ...rest.purposes,
+        value: rest.purposes.value ?? [],
       },
       produces: {
+        ...produces,
         value: {
           ...producible,
-          id: produces.p.properties.id,
+          id: pr.p.properties.id,
           __typename: (ProducibleType as any)[typeName],
```

Comment:
```suggestion
          __typename: typeName,
```
With comment on `typeName` declaration.

Reply:
`difference` still returns string not ProducibleType value. 
https://i.imgur.com/HtRnxtM.png

Reply:
https://i.imgur.com/zxgGObJ.png
This is working fine.